src/renderer/components/Map.js

```diff
@@ -0,0 +1,93 @@
+const React = require('react')
+const { ipcRenderer } = require('electron')
+const mapboxgl = require('mapbox-gl')
+const geojsonExtent = require('@mapbox/geojson-extent')
+const moment = require('moment')
+const MapLayerFactory = require('./helpers/MapLayerFactory')
+
+class Map extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      points: []
+    }
+    this.initialTimeOffset = 48 // hours
+    this.renderLayer = this.renderLayer.bind(this)
+    this.onMapClick = this.onMapClick.bind(this)
+  }
+
+  componentDidMount () {
+    this.componentDidMount = Date.now()
+    console.log('componentDidMount', this.componentDidMount)
+    mapboxgl.accessToken = MapLayerFactory.getAccessToken()
+    this.map = new mapboxgl.Map(
+      {
+        container: 'map',
+        style: 'mapbox://styles/mapbox/streets-v8',
+        zoom: 4,
+        center: [8, 48],
+        attributionControl: false
+      })
+    this.map.on('load', this.renderLayer)
+    this.map.on('click', this.onMapClick)
+  }
+
+  renderLayer () {
+    const { selectedChat } = this.props
+    const contacts = selectedChat.contacts
+    let allPoints = []
+
+    console.log('map loaded', this.componentDidMount - Date.now())
+    let locationsForChat = ipcRenderer.sendSync('getLocations', selectedChat.id, 0, moment().unix() - (this.initialTimeOffset * 3600), 0)
+    contacts.map(contact => {
+      let locationsForContact = locationsForChat.filter(location => location.contactId === contact.id)
+      if (locationsForContact && locationsForContact.length) {
+        let pointsForLayer = locationsForContact.reduce(
+          (points, point) => {
+            if (point.acc > 0) {
+              points.push([point.lon, point.lat])
+            }
+            return points
+          },
+          []
+        )
+        this.map.addLayer(MapLayerFactory.getGeoJSONLineLayer(pointsForLayer, contact))
+        this.map.addLayer(MapLayerFactory.getGeoJSONPointsLayer(locationsForContact, contact))
+
+        let lastPoint = locationsForContact[locationsForContact.length - 1]
+        let lastDate = moment(lastPoint.tstamp * 1000).format('Y-m-d LT')
```

Comment:
I think you can use the localized time string for this. have a look at `renderer/conversations/{formatRelativeTime|Timestamp}.js`.